src/django_components/component_registry.py

```diff
@@ -237,7 +244,19 @@ def __init__(
 
         ALL_REGISTRIES.append(cached_ref(self))
 
+        extensions.on_registry_created(
+            OnRegistryCreatedContext(
+                registry=self,
+            )
+        )
+
     def __del__(self) -> None:
+        extensions.on_registry_deleted(
```

Comment:
Since there are now the hooks that are triggered on Component / Registry creation or deletion, I had to also make use of the `__del__` method here ([and here](https://github.com/django-components/django-components/pull/1009/files#diff-35c84ebceae50f7149f03bbcaf2b4f1289bc2984bf7859ca521b5dbe7960c396R206))